### PR TITLE
improve migration script

### PR DESCRIPTION
**Pull request by jellegerbrandy**
Fix: #34

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ RUN chmod +x /wait /entry
 
 WORKDIR /usr/app
 COPY . .
-RUN npm ci
+RUN npm install
```
Comment (orenyodfat) [inline Dockerfile]: why is that changed back ?

Comment (jellegerbrandy) [reply Dockerfile]: It is changed back because I'd like to buidl the container in the `node_modules/@daostack/subgraph` repo, but the intallation of npm does not include the `package-lock.json`. And so `npm ci` fails. Unfortunately, because it would be better/more predictable.

Comment (tsuberim) [reply Dockerfile]: @jellegerbrandy We should instead publish the image to dockerhub and use it in the `docker-compose.yml` to both avoid this problem and shorten build times.

File: docker-compose.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ services:
       - ganache
     environment:
       ethereum: http://ganache:8545
-      test_mnemonic: behave pipe turkey animal voyage dial relief menu blush match jeans general
+      test_mnemonic: myth like bonus scare over problem client lizard pioneer submit female collect
```
Comment (orenyodfat) [inline docker-compose.yml]: just for curiosity ...why this was changed?

Comment (ben-kaufman) [reply docker-compose.yml]: I think it's because the new one is the default mnemonic of Ganache.

File: Dockerfile (hunk 1 above)
Comment (jellegerbrandy) [reply Dockerfile]: Yes I agree.

File: docker-compose.yml (hunk 2 above)
Comment (jellegerbrandy) [reply docker-compose.yml]: This is the mnemonic that gnereated the accoutns of ganache when it runs in `--deterministic` mode but without inventing our own test mnemonic. https://github.com/daostack/devops#ganache

Comment (jellegerbrandy) [reply docker-compose.yml]: (see https://github.com/daostack/subgraph/pull/32/files#diff-4e5e90c6228fd48698d074241c2ba760L50)

File: ops/deployDaoStack.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+const DAOstackMigration = require('@daostack/migration');
```
Comment (jellegerbrandy) [inline ops/deployDaoStack.js]: this script replaces `ops/migration.js` and uses the scripts from the mgration repo instead

File: ops/generate-subgraph.js [hunk 4]
```diff
@@ -16,11 +17,17 @@ async function generateSubgraph() {
 	const dataSources = files.map(file => {
 		const contract = path.basename(path.dirname(file));
 		const { abis, entities, eventHandlers } = yaml.safeLoad(fs.readFileSync(file, 'utf-8'));
+
+		const contractAddress = addresses.private.base[contract] || addresses.private.dao[contract];
+
+		if (!contractAddress) {
+			throw Error(`Address for contract ${contract} not found in ${migrationFile}`);
```
Comment (jellegerbrandy) [inline ops/generate-subgraph.js]: let's try to do explicit error handling in our scripts so we catch errors early

File: Dockerfile (hunk 1 above)
Comment (jellegerbrandy) [reply Dockerfile]: Actually, you are just using this container to run the tests in, right? (Is that a good idea?) In that case, there would be no need to build that container at all...

File: src/mappings/Avatar/mapping.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import 'allocator/arena';
-export { allocate_memory };
+//// export { allocate_memory };
```
Comment (orenyodfat) [inline src/mappings/Avatar/mapping.ts]: please remove this comments

File: src/mappings/Avatar/mapping.ts [hunk 6]
```diff
@@ -18,7 +18,7 @@ function handleAvatarBalance(
 
   let avatar = store.get('AvatarContract', address.toHex()) as AvatarContract;
   if (avatar == null) {
-    avatar = new AvatarContract();
+    avatar = new AvatarContract(address.toHex());
     avatar.id = address.toHex();
```
Comment (orenyodfat) [inline src/mappings/Avatar/mapping.ts]: remove redundant.id - this could be done in other pr .

File: ops/deployDaoStack.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  options = {
+    // web3 provider url
+    provider,
+    quiet: true,
+    // disable confirmation messages
+    force: true,
+    // filepath to output the migration results
+    output: path.resolve(migrationFileLocation),
+    // private key of the account used in migration (overrides the 'mnemonic' option)
+    // privateKey: '0x8d4408014d165ec69d8cc9f091d8f4578ac5564f376f21887e98a6d33a6e3549',
```
Comment (orenyodfat) [inline ops/deployDaoStack.js]: do we need this comment here ? @jellegerbrandy

Comment (jellegerbrandy) [reply ops/deployDaoStack.js]: no we do not :-)
